### PR TITLE
Split development documentation into several pages

### PR DESCRIPTION
**Pull request by nirname**
## :bookmark_tabs: Summary

Simplifying development documentation

Related:
* #2910
* #4596

## :straight_ruler: Design Decisions

Current development pages were far to long to read and understand:

* Split development page into several pages, previous one was far too big to read it briefly
* Nested sections at the left menu buttons caused Prev Page and Next Page buttons to be wrong
* Sections inside development pages are 2nd level now (heading 2) and visible at the right menu 
* Added some missing commands to start development
* Removed links to obsolete `n00b-overview` page and the page itself
* Added documentation for development in Docker and cross links between host and docker installation guides
* Fixed top menu (added link to contribution sections)
* Squashed some sections of introduction page together as they repeated each other
* Some of the headings changed their levels to more consistent structure

Other things will be done in upcoming requests such as:
* Removing `n00b` from file names
* Making shorter version of development pages (aka Quick Start Guide) and updating `CONTRUBITION.md`
* Adding section 'Why use Mermaid'

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [x] :computer: have added necessary unit/e2e tests.
- [x] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/docs/community/development.md#3-update-documentation) is used for all new features.
- [x] :bookmark: targeted `develop` branch

### CODE REVIEW

**Comment by nirname** [comment]
@Yokozuna59 I ensured that "Adding new diagram" is a separate file, and a separate line in the left menu. Editing it is harmless, there is no soon changes to this file in this or upcoming PR's. Edit it should you need to

File: packages/mermaid/src/docs/community/documentation.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+```danger
+Danger content
+```
+
+````
+
+```note
+If the change is _only_ to the documentation, you can get your changes published to the site quicker by making a PR to the `master` branch.
```
Comment (sidharthv96) [inline packages/mermaid/src/docs/community/documentation.md:51]: In that case, your branch should be based on `master`, not `develop`.

**Review by sidharthv96 — CHANGES_REQUESTED** [review]

**Review by aloisklink — COMMENTED** [review]
Thanks for splitting up the https://github.com/mermaid-js/mermaid/pull/4596 PR! This is a lot easier to review :smile:

Other than @sidharthv96's comments, it looks good to me, so I'll let @sidharthv96 officially "Approve" this PR when it's done.

Btw, for future PRs (it's not worth doing for this PR), it would be even easier to review if you split up the commits where you
  - move lines between files and where you 
  - modify the lines.

That way us reviewers can easily see which lines you've just copied over, and we can ignore them, and only focus on the lines that have changed (you can then even write in your PR description to tell us to ignore some commits :wink:)

---

Hmmm, weird, I swear `docs/*` is in the PR labeler config, but for some reason the **Area: Documentation** label didn't get automatically applied.

https://github.com/mermaid-js/mermaid/blob/b85c011cd174b2a0f3653c60ba03c05cca42a32c/.github/pr-labeler.yml#L4

Maybe the labeller action doesn't work on PRs from forks? Oh well, it's not too difficult to manually label PRs.

File: packages/mermaid/src/docs/.vitepress/theme/redirect.ts [hunk 2]
```diff
@@ -50,6 +50,7 @@ const idRedirectMap: Record<string, string> = {
   'n00b-advanced': 'config/n00b-advanced',
   'n00b-gettingstarted': 'intro/n00b-gettingStarted',
   'n00b-overview': 'community/n00b-overview',
+  'community/n00b-overview': 'intro/n00b-gettingStarted',
   'n00b-syntaxreference': 'intro/n00b-syntaxReference',
```
[the remark this answers is not in the document]
Comment (sidharthv96) [reply packages/mermaid/src/docs/.vitepress/theme/redirect.ts:54]: Yes. But an entry has to be added to `urlRedirectMap` too`, as we are changing a page in the current site. 

I have added some comments in cfffba28177505384ff124ce7477c4b30f1e7555 to clarify.

File: packages/mermaid/src/docs/intro/index.md [hunk 3]
```diff
@@ -99,6 +99,7 @@ To Deploy Mermaid:
 - [Mermaid CLI](https://github.com/mermaid-js/mermaid-cli)
 - [Mermaid Webpack Demo](https://github.com/mermaidjs/mermaid-webpack-demo)
 - [Mermaid Parcel Demo](https://github.com/mermaidjs/mermaid-parcel-demo)
+- [HTTP Server](https://github.com/TomWright/mermaid-server)
```
[the remark this answers is not in the document]
Comment (nirname) [reply packages/mermaid/src/docs/intro/index.md:102]: Yep, they are already on the main page, and I think it's better to move this

File: packages/mermaid/src/docs/community/questionsAndSuggestions.md
[the remark this answers is not in the document]
Comment (nirname) [reply packages/mermaid/src/docs/community/questionsAndSuggestions.md:1]: emmmm, there are plenty of other names that are not kebab-case, are we planning to rename them as well?

File: packages/mermaid/src/docs/community/questionsAndSuggestions.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+This helps the team know the relative interest in something and helps them set priorities and assignments.
+
+Feel free to add to the discussion on the issue or topic.
+
+If you can't find anything that already addresses your question or suggestion, _open a new issue:_
+
+Log in to [GitHub.com](https://www.github.com), open or append to an issue [using the GitHub issue tracker of the mermaid-js repository](https://github.com/mermaid-js/mermaid/issues?q=is%3Aissue+is%3Aopen+label%3A%22Area%3A+Documentation%22).
+
+## How to Contribute a Suggestion
```
[the remark this answers is not in the document]
Comment (nirname) [reply packages/mermaid/src/docs/community/questionsAndSuggestions.md:22]: There is a TODO already, at the very beginning of a file. This is a copy and paste from previous page. Do we need another one?

File: docs/community/docker-development.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+Now you are ready to make your changes! Edit whichever files in `src` as required.
+
+Open <http://localhost:9000> in your browser, after starting the dev server.
+There is a list of demos that can be used to see and test your changes.
+
+If you need a specific diagram, you can duplicate the `example.html` file in `/demos/dev` and add your own mermaid code to your copy.
+
+That will be served at <http://localhost:9000/dev/your-file-name.html>.
+After making code changes, the dev server will rebuild the mermaid library. You will need to reload the browser page yourself to see the changes. (PRs for auto reload are welcome!)
+
```
[the remark this answers is not in the document]
Comment (nirname) [reply docs/community/docker-development.md:92]: This is simply not true
https://github.com/mermaid-js/mermaid/pull/4744/files#diff-daf94d9ac125e821ddfd6f7341cf6b3c79f5e39d781ddcc32a1c5dcd356419bdR68-R76

File: packages/mermaid/src/docs/community/questionsAndSuggestions.md
[the remark this answers is not in the document]
Comment (sidharthv96) [reply packages/mermaid/src/docs/community/questionsAndSuggestions.md:1]: Just making sure we are not adding new files. Renaming other files will mean adding more redirects and all, which is extra work without much benefits.

File: docs/community/docker-development.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (sidharthv96) [reply docs/community/docker-development.md:92]: My bad, I searched for `example.html`, but the other file was collapsed by github. So only this matched.

**Review by sidharthv96 — APPROVED** [review]

File: packages/mermaid/src/docs/community/questionsAndSuggestions.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sidharthv96) [reply packages/mermaid/src/docs/community/questionsAndSuggestions.md]: I wasn't sure if it was actual TODO, or if you missed a part when copy pasting. Just confirmed that.

**Pull request merged**